lib/point_quest/quests/quest.ex

```diff
@@ -179,4 +192,13 @@ defmodule PointQuest.Quests.Quest do
       {:error, :round_not_active}
     end
   end
+
+  def handle(%Commands.AddSimpleObjective{} = command, quest) do
+    if Enum.any?(quest.objectives, fn objective -> objective.id == command.quest_objective end) do
+      {:error, :objective_already_exists}
+    else
+      objective = Questable.to_objective(command) |> Ecto.embedded_dump(:json)
+      {:ok, Event.ObjectiveAdded.new!(%{quest_id: command.quest_id, objective: objective})}
```

Comment:
I'm unsure if this is the appropriate place for this. It felt a little weird to try handling this in the command, but it also feels weird having it here in the handle (little bit of separation of concerns anxiety, but maybe I'm just being overly sensitive)

Reply:
Open to suggestions on better pattern/placement for this functionality.

Reply:
Doing the check to se if the objective is unique? I think anything to do with the current state of the model and whether the change would result in a bad state should happen in the model. So I think this is a good place.